### PR TITLE
Stop testing compatibility against Databricks 7.3

### PR DESCRIPTION
**Pull request by findinpath**
## Description
Databricks 7.3 is reaching on September 2023 end of support.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by findepi** [comment]
some checks have failed

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java [hunk 1]
```diff
@@ -71,7 +70,7 @@ public void testAddColumnWithCommentOnTrino()
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, DELTA_LAKE_EXCLUDE_113, PROFILE_SPECIFIC_TESTS})
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, DELTA_LAKE_EXCLUDE_104, DELTA_LAKE_EXCLUDE_113, PROFILE_SPECIFIC_TESTS})
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlterTableCompatibility.java:73]: optional: This test class contains some comments about version 7.3. We could update/remove those comments.

**Review by ebyhr — COMMENTED** [review]
Please remove `7.3 LTS` from delta-lake.rst:
```
* Tables written by Databricks Runtime 7.3 LTS, 9.1 LTS, 10.4 LTS, 11.3 LTS, and
  12.2 LTS are supported.
```

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 2]
```diff
@@ -17,7 +17,7 @@ Requirements
 
 To connect to Databricks Delta Lake, you need:
 
-* Tables written by Databricks Runtime 7.3 LTS, 9.1 LTS, 10.4 LTS, 11.3 LTS, and
```
Comment (findepi) [inline docs/src/main/sphinx/connector/delta-lake.rst:20]: While at it, would it make sense to capture some tables created by 7.3 as static test resources and run tests against them?

Then we wouldn't change this file. Of course it's less about changing this file, but more about staying compatible with data out there. We should be able to read any Delta table, including those written by DBR << 7.

Comment (findinpath) [reply docs/src/main/sphinx/connector/delta-lake.rst:20]: BTW, I think most of the initial tables from the test resources used for testing `trino-delta-lake` are actually created with Databricks 7.3.

Trino can likely read as well tables written with non-LTS  version of Databricks, but we're not mentioning this here.
I think there's no loss / misleading information if we remove `7.3 LTS` from the documentation.

Comment (findepi) [reply docs/src/main/sphinx/connector/delta-lake.rst:20]: I care less about docs, and more about testing coverage.
Tables created by DBR 7.3 do exist in the world and we must keep supporting them.
Is this test covered?

**Comment by findinpath** [comment]
@findepi please run the PR with secrets (I've slipped in a change which enables an existing product tests).

**Comment by ebyhr** [comment]
/test-with-secrets sha=ca1b3e7f8795a7c59d60b2bb41f2aabc3ed65069

**Comment by findinpath** [comment]
I guess that the failure [pt (default, suite-delta-lake-databricks73, )](https://github.com/trinodb/trino/actions/runs/5850234042/job/15859665647#logs) has to do with the fact that `test-with-secrets` functionality doesn't cope well with changes on `ci.yml` file.

**Comment by ebyhr** [comment]
Sent #18665 within repository.

**Comment by findinpath** [comment]
Rebased on `master` to address conflicts

**Comment by findepi** [comment]
/test-with-secrets sha=7cb65b2d1451359ce579e76fa22fa5c507e6f0e4

**Comment by ebyhr** [comment]
Updated #18665 because the `/test-with-secrets` command failed because of `ci.yml` change.

**Comment by findinpath** [comment]
Unrelated test infra failure:
`ci / pt (default, suite-3)` failed because not being able to retrieve the docker image from  ghcr.io
https://github.com/trinodb/trino/actions/runs/6027449276/job/16353173416?pr=18538

```
2023-08-30T16:51:12.761Z	ERROR	main	io.trino.tests.product.launcher.cli.TestRun	Failure
org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=ghcr.io/trinodb/testing/hdp2.6-hive-kerberized:81, imagePullPolicy=io.trino.testing.containers.ConditionalPullPolicy@4978777f, imageNameSubstitutor=org.testcontainers.utility.ImageNameSubstitutor$LogWrappedImageNameSubstitutor@4215e133)
```

**Pull request merged**